app/src/main/java/org/oppia/android/app/devoptions/marktopicscompleted/MarkTopicsCompletedViewModel.kt

```diff
@@ -61,18 +55,19 @@ class MarkTopicsCompletedViewModel @Inject constructor(
     return allTopics.getOrDefault(mutableListOf())
   }
 
-  private fun processAllTopics(allTopics: List<Topic>): List<TopicSummaryViewModel> {
-    val itemList = mutableListOf<TopicSummaryViewModel>()
-    availableTopicIdList.clear()
+  private fun processAllTopics(allTopics: List<Topic>): List<TopicViewModel> {
+    itemList.clear()
     allTopics.forEach { topic ->
       val isCompleted = modifyLessonProgressController.checkIfTopicIsCompleted(topic)
-      itemList.add(TopicSummaryViewModel(topic, isCompleted))
-      if (!isCompleted) availableTopicIdList.add(topic.topicId)
+      itemList.add(TopicViewModel(topic, isCompleted))
     }
     return itemList
   }
 
-  fun setInternalProfileId(internalProfileId: Int) {
-    this.internalProfileId = internalProfileId
+  fun setProfileId(profileId: ProfileId) {
+    this.profileId = profileId
   }
+
+  /** Returns a list of [TopicViewModel] whose progress can be modified. */
```

Comment:
```suggestion
  /** Returns a list of [TopicViewModel]s whose progress can be modified. */
```

Ditto elsewhere (where applicable).

Reply:
Done.